### PR TITLE
Fix ARK status error handling

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

Fixes wrong error handling behavior at ARK enablement check. A missing DB record is not an error that we should propagate.

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)

### CODE REVIEW

File: api/ark/backupservice/status.go [hunk 1]
```diff
@@ -30,12 +30,15 @@ import (
...
 	logger := correlationid.Logger(common.Log, c)
-	logger.Info("checking ARK deployment status")
+	logger.Debug("checking ARK deployment status")
 
 	schedulesSvc := common.GetARKService(c.Request).GetSchedulesService()
 	_, err := schedulesSvc.List()
 	if err != nil {
-		err = errors.New("backup service not deployed")
+		err = errors.WrapIf(err, "backup service not deployed")
+		if !gorm.IsRecordNotFoundError(errors.Cause(err)) {
```
Comment (sagikazarmark) [inline api/ark/backupservice/status.go]: How about checking the error first, before wrapping it?

File: api/ark/backupservice/status.go [hunk 2]
```diff
@@ -46,13 +49,15 @@ func StatusDeprecated(c *gin.Context) {
...
-	logger.Info("checking ARK deployment status")
+	logger.Debug("checking ARK deployment status")
 
 	schedulesSvc := common.GetARKService(c.Request).GetSchedulesService()
 	_, err := schedulesSvc.List()
 	if err != nil {
-		err = emperror.Wrap(err, "backup service not deployed")
-		common.ErrorHandler.Handle(err)
+		err = errors.WrapIf(err, "backup service not deployed")
+		if !gorm.IsRecordNotFoundError(errors.Cause(err)) {
```
Comment (sagikazarmark) [inline api/ark/backupservice/status.go]: See the comment above

File: api/ark/backupservice/status.go (hunk 1 above)
Comment (waynz0r) [reply api/ark/backupservice/status.go]: 👍

File: api/ark/backupservice/status.go (hunk 2 above)
Comment (waynz0r) [reply api/ark/backupservice/status.go]: 👍

**Pull request merged**